File: perl/expreg_args/expreg.pl
#!/usr/bin/perl
=begin comment
=end comment
=cut
sub match{
    $str = $_[0];
    if($ARGV[0] =~ /^[A-Z]([a-z]|ñ){1,5}$/){
	return 1;
    }
    return 0;
}

if($#ARGV + 1 != 1){
    print "Recibe un solo argumento\nDebe ser de la forma : \n";
} else {
    if(match($ARGV[0])){
	print "Felicidades la cadena es valida!\n";
    } else {
	print "La cadena no es correcta\n";
    }
}
print "Cerrando programa ...\n"

File: julia/expreg_args/expreg.jl
#!/usr/bin/julia

#=
 * Verifica que la cadena word cumpla con la expresión regular
 *
 * @param word  cadena a validar
 *
 * @return  true si es vallida, false en otro caso
=#
function match(word)
  return ismatch(r"^[A-Z]([a-z]|ñ){1,5}$", word) # verifica que la cadena sea valida
end

if endof(ARGS) != 1
  println("Recibe un solo argumento\nDebe ser de la forma :")
else
  if match(ARGS[1])
    println("Felicidades la cadena es valida!")
  else
    println("La cadena no es correcta")
  end
end
println("Cerrando programa ...")
